### DISCUSSION TITLE
Importing GPX into Garmin BaseCamp gives messy result

### DISCUSSION THREAD

**Discussion by jps-aldridge** [waymarkedtrails/waymarked-trails-site#433] [Q&A] [answered]
If I download a GPX file from waymarkedtrails and import it into the Garmin BaseCamp application I get a very messy result.
![BaseCamp](https://user-images.githubusercontent.com/34809163/223107311-a1669ab5-4b4d-4cdf-a66e-33212c7ddabe.jpg)

If I import the GPX file into Google Earth it looks much better.
![GoogleEarth](https://user-images.githubusercontent.com/34809163/223109454-675a792e-b5b5-4217-a6c8-c8213fbd0152.jpg)

These examples are from the Harcamlow Way trail, [https://hiking.waymarkedtrails.org/#route?id=3318993](url)

Is there a easy way of tidying up the BaseCamp version? Thanks for any advice!

**Comment by lonvia** [comment, marked as answer]
See #268, #414.

**Comment by jps-aldridge** [comment]
Thank you! I did search on the issues list, but I didn't find these because these issues are marked "Closed".

Is there the necessary sorting/order handling planned anytime soon? Or should I maybe try to write some code to try to repair the GPX file?

**Comment by jps-aldridge** [comment]
I wrote some Python code to tidy up a GPX file: here it is, in case it's useful to anyone else.
[repairgpx.py.txt](https://github.com/waymarkedtrails/waymarked-trails-site/files/10908835/repairgpx.py.txt)
As well as having Python, you'll need to install the gpxpy package.
